source/Ship.h

```diff
@@ -703,11 +739,11 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// Accrued "discharge damage" that will affect this ship's shields over time.
 	double discharge = 0.;
 	// Accrued "corrosion damage" that will affect this ship's hull over time.
-	double corrosion = 0.;
+	DamageLog corrosion;
 	// Accrued "leak damage" that will affect this ship's fuel over time.
 	double leakage = 0.;
 	// Accrued "burn damage" that will affect this ship's heat over time.
-	double burning = 0.;
+	DamageLog burning;
```

Comment:
Unless I'm blind, you're never checking the source of this particular damage log, so it should just be a normal double still.